openedx_blockstore_relay/transfer_data.py

```diff
@@ -0,0 +1,257 @@
+        Upload any files found in the filesystem.
+        """
+        for item in filesystem.walk():
+            for unit_file in item.files:
+                file_path = os.path.join(item.path, unit_file.name)
+                self._add_bundle_file(
+                    path=item.path,
+                    name=unit_file.name,
+                    data=filesystem.open(file_path, 'rb'),
+                    scrape_assets=True,
+                )
+
+    def upload_olx(self, olx_node, block, store_component=False):
+        """
+        Upload the OLX file to the bundle, and store it in the manifest if requested.
+        """
+        block_id = block.location.block_id
+        block_type = block.location.block_type
+        return self._add_bundle_file(
+            path=os.path.join(os.sep, block_type),
+            name='.'.join([block_id, 'olx']),
+            data=ElementTree.tostring(olx_node, encoding="utf-8"),
+            content_type=self.olx_type(block_type),
+            store_component=store_component,
+            scrape_assets=True,
+        )
+
+    def upload_assets(self):
+        """
+        Upload the collected static asset files to the bundle.
+        """
+        for (asset_path, asset_uri) in self.assets:
+            asset_url = urljoin(self.asset_root_url, asset_uri)
+            try:
+                # Stream the asset to the bundle file POST
+                get_asset = requests.get(asset_url, stream=True)
+                get_asset.raise_for_status()
+                get_asset.raw.decode_content = True
+                self._post_bundle_file(
+                    path=asset_path,
+                    data=get_asset.raw,
+                    content_type=get_asset.headers['Content-Type'],
+                )
+            except HTTPError as exc:
+                LOG.error("Unable to fetch static asset (%s): %s", asset_url, exc)
+
+    def upload_manifest(self, name='bundle.json', path=os.sep):
+        """
+        Upload the bundle manifest file.
+        """
+        return self._post_bundle_file(
+            path=path,
+            name=name,
+            data=json.dumps(self.manifest, ensure_ascii=False),
+            content_type='application/json',
+        )
+
+    def _add_bundle_file(self,
+                         data,
+                         path,
+                         name=None,
+                         content_type='application/octet-stream',
+                         store_component=False,
+                         scrape_assets=False):
+        """
+        Add the given file data to the bundle.
+
+        Scrape the static assets and update the manifest if requested.
+        """
+        if scrape_assets and hasattr(data, 'read'):
+            # Read the data from the file pointer into a string so we can scrape it later
+            data = data.read()
+
+        response = self._post_bundle_file(path=path, name=name, data=data, content_type=content_type)
+
+        if store_component:
```

Comment:
This should happen in `upload_olx()` for the root `usage_key` that was passed to the command.